server/src/main/java/org/elasticsearch/common/settings/AbstractScopedSettings.java

```diff
@@ -956,6 +956,51 @@ public Settings archiveUnknownOrInvalidSettings(
         }
     }
 
+    /**
+     * Deletes invalid or unknown settings. Any setting that is not recognized or fails validation
+     * will be deleted. This behaviour is desired when dealing with unknown index settings on
+     * system indices.
+     *
+     * @param settings        the {@link Settings} instance to scan for unknown or invalid settings
+     * @param unknownConsumer callback on unknown settings (consumer receives unknown key and its
+     *                        associated value)
+     * @param invalidConsumer callback on invalid settings (consumer receives invalid key, its
+     *                        associated value and an exception)
+     * @return a {@link Settings} instance with the unknown or invalid settings removed
+     */
+    public Settings deleteUnknownOrInvalidSettings(
+        final Settings settings,
+        final Consumer<Map.Entry<String, String>> unknownConsumer,
+        final BiConsumer<Map.Entry<String, String>, IllegalArgumentException> invalidConsumer
+    ) {
+        Settings.Builder builder = Settings.builder();
+        boolean changed = false;
+        for (String key : settings.keySet()) {
+            try {
```

Comment:
It looks to me like line 982 is what we're relying on to throw the IllegalArgumentException. Would it make sense to scope this try/catch block more tightly, or to add a comment on that line that `setting.get(settings)` will throw an IllegalArgumentException if a setting has an invalid value?

Reply:
Yeah, I was following what was done for archive. I'll add a related comment.